easybuild/tools/options.py

```diff
@@ -1581,6 +1590,17 @@ def set_up_configuration(args=None, logfile=None, testing=False, silent=False):
         'try_to_generate': try_to_generate,
         'valid_stops': [x[0] for x in EasyBlock.get_steps()],
     }
+
+    # Remove existing singletons if reconfigure==True (allows reconfiguration when looping over EasyStack items)
+    if reconfigure:
+        BuildOptions.__class__._instances.clear()
+    elif len(BuildOptions.__class__._instances) > 0:
+        log.warn("set_up_configuration is about to call init() and init_build_options(). "
```

Comment:
better use `print_warning` here, otherwise the warning will only appear in the log file

```python
msg = '\n'.join([
    "...",
    "...",
])
print_warning(msg, log=log)
```

Reply:
Done

Reply:
fixed in d1298598e45abea579ae25c8a1d00ab827ceee89